dts/bindings/gpio/stemma-qt-connector.yaml

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2024 TOKITA Hiroshi
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    STEMMA QT is a 1.0mm-JST-SH connector for I2C devices.
+    The pin assignments are as follows.
+
+        0  SCL
+        1  SDA
+        -  VCC (3-5V)
+        -  GND
+
+    STEMMA, Grove, Qwiic, and Gravity connectors are similar to this,
+    but the connector form and VCC voltage are different.
+    Please see
+    https://learn.adafruit.com/introducing-adafruit-stemma-qt/stemma-qt-comparison#quick-comparison-3035247
+
+compatible: "stemma-qt-connector"
```

Comment:
This point will likely cause some debate, but I think keeping the brands and trademarks as they are is best.
This is a kind of guarantee of interoperability, so I think following it would be less confusing.

As I wrote in the commit message, STEMMA QT is only for I2C, and this yaml file is merely a formal definition and doesn't practically have much meaning.
As for STEMMA QT, I think interoperability will be ensured by the naming convention for the labels assigned to i2c.

STEMMA QT and Qwiic have an inclusion relationship in their specifications, and STEMMA QT includes the Qwiic specifications.

In such cases, I think it would be better to create two aliases on the board side, that is,

```
stemma_qt_i2c: &i2c0 { }
qwiic_qt_i2c: &i2c0 { }
```

like this.

In software terms, the signals are in the same order, so it would be easier to unify them, including Grove. Still, I think it would be better to use their official names to identify them and make clear their interoperability.

Reply:
does not seem like doubling up is worth it, I mean a shield will be either stemmma/qwiic or grove or the other one but the stemma/qwiic ones interoperate, let's pick one and go with it, really either is good

or rather is either is bad, can't believe they did all this work to do the branding and didn't think about adding an interrupt line to the connector, anyway...

Reply:
Qwiic is 3.3v, and Stemma is 3V - 5V, so the difficulty is that they are only partially "interoperable". Strictly speaking, qwiic devices can be connected to stemma buses, but Stemma devices cannot be connected to qwiic buses. Also, by definition, a 5V stemma bus can be considered, so there is no interoperability with the Qwiic device in this case. It is a challenging task to express that it will work properly if it can be connected (if it can be compiled in software terms).

Since this PR concerns only the device side, I think the current definition is fine.

> or rather is either is bad, can't believe they did all this work to do the branding and didn't think about adding an interrupt line to the connector, anyway...

This is it!